spec/formatting.md

```diff
@@ -18,3 +18,17 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Bidirectional Isolation
+
+Where appropriate,
```

Comment:
I agree that "Where appropriate ... " is far too vague to be a conformance MUST. 

However, condition is too strong. 
1. There are many instances where isolation is not required, such as a single Arabic word in a placeholder surrounded by Latin. It would be better to supply a functional description. Off the top of my head: w
    - Where the text of the placeholder or surrounding text would rearrange due to the insertion.
2. There is a significant number of products in use that do not yet support isolates. [Aside: I really wish I'd made the embeddings be isolating in the first place, but water far under the bridge.] For use of MF2.0 with those products, we have to have a fallback position (ugly as that is).

Reply:
I've updated the text following @aphillips's suggestions.

@macchiati Is there an existing algorithm for determining when isolation would not be required? I presume that this would require scanning both the placeholder's stringified value as well as that of its surroundings, and ensuring that neither have any weakly typed characters immediately adjacent to the other.

I would prefer an algorithm that would not require such introspection, even if it does come at a cost of needing to include isolating characters in some situations where they have no effect.

To support products and situations which do not support isolates, would it be sufficient to mandate the default behaviour, and allow for a formatter to provide alternatives? For example, in the JS implementation, I could imagine including an option like:

```ts
type MesageFormatOptions = {
  isolate = 'isolate' | 'embed' | 'none' = 'isolate',
  ...
}
```

This would allow for defaulting to isolation (explicitly spec-defined), while also being able to provide implementation-defined fallback solutions.